eng/FlakyTests.BeforeArcade.props

```diff
@@ -0,0 +1,18 @@
+<Project>
+  <!-- Local Dev Flakiness -->
+  <PropertyGroup>
+    <_FlakyRunAdditionalArgs>-trait "Flaky:All=true"</_FlakyRunAdditionalArgs>
+    <_NonFlakyRunAdditionalArgs>-notrait "Flaky:All=true"</_NonFlakyRunAdditionalArgs>
+  </PropertyGroup>
+
+  <!-- Azure Pipelines Flakiness -->
+  <PropertyGroup Condition="'$(AGENT_OS)' != ''">
+    <_FlakyRunAdditionalArgs>$(_FlakyRunAdditionalArgs) -trait "Flaky:AzP:All=true" -trait "Flaky:AzP:OS:$(AGENT_OS)=true"</_FlakyRunAdditionalArgs>
+    <_NonFlakyRunAdditionalArgs>$(_NonFlakyRunAdditionalArgs) -notrait "Flaky:AzP:All=true" -notrait "Flaky:AzP:OS:$(AGENT_OS)=true"</_NonFlakyRunAdditionalArgs>
```

Comment:
Aren't `-trait "Flaky:AzP:All=true"` and `-notrait "Flaky:AzP:All=true"` unnecessary in these two lines?

Reply:
Not quite. The logic isn't as simple as just setting the OS-based flag. The `Flaky` attribute is trying to be a little agnostic as to the "filters" in use (since we may add other pivots like architectures, etc.), so if you say it's flaky on Darwin, like so:

```csharp
[Flaky("...", AzurePipelines.macOS)]
```

Then we set the xUnit trait: `Flaky:AzP:OS:Darwin=true`

If you say it's flaky everywhere, like so:

```csharp
[Flaky("...", AzurePipelines.All)]
```

Then we set a **different** xUnit trait: `Flaky:AzP:All=true` since we don't want the `Flaky` attribute logic to have to enumerate every possible pivot and filter.

Reply:
I did it this way so that the complexity could live in the build script rather than having to keep updating the `Flaky` attribute whenever we add new pivots or new pivot pieces. If we wanted to simplify this logic, we'd need `Flaky` to inject traits representing all possible pivots for which the test is flaky which makes things like `AzurePipelines.All` tricky.